discovery/syncer.go

```diff
@@ -906,6 +935,17 @@ func (g *GossipSyncer) processChanRangeReply(msg *lnwire.ReplyChannelRange) erro
 	return nil
 }
 
+// isTimestampSane is a helper function that can be used to determine if a
+// timestamp sent in a ReplyChannelRange message is considered sane.
+func isTimestampSane(t time.Time) bool {
+	// TODO(elle): Are there more logical bounds we can set here? Could we
+	//  perhaps check that the timestamps they send match the timestamps of
+	//  the full messages that they send?
+
+	// A timestamp in the future is not considered sane.
+	return !t.After(time.Now())
```

Comment:
I'm not sure if this works, what if there is clock drift? I'm also not sure how useful this sanity check is -- what's the worst an attacker can do without the sanity check?

Reply:
AFAICT, the worst an attacker can do is trick us into unnecessarily requesting *all* channel updates within the range.  Considering that would put some load on the attacker themselves to send those updates, I don't think it's much of an attack.

Either way, this simple sanity check can be trivially defeated by using `time.Now()` as the timestamp.

Reply:
> Considering that would put some load on the attacker themselves to send those updates, I don't think it's much of an attack.

good point. 

removed 👍